### ISSUE TITLE
Keypoints RCNN with 3 keypoints and My Dataset

### ISSUE THREAD

**Issue by hkwei** [pytorch/vision#2125]
## ❓Zero  loss_keypoint ,and zero  loss_classifier:

I defined a  model :
```python
    model = torchvision.models.detection.keypointrcnn_resnet50_fpn(pretrained=True)
    model.roi_heads.box_predictor = FastRCNNPredictor(1024, 2)
    keypoint_predictor = KeypointRCNNPredictor(512, num_keypoints=num_keypoints)
    model.num_keypoints = 3
    model.roi_heads.keypoint_predictor = keypoint_predictor
    return model
```
a script to test the input and output of model in train mode:
```python
images,targets = next(iter(data_loader))
print(targets)
images = list(image for image in images)
targets = [{k: v for k, v in t.items()} for t in targets]
output = model(images,targets)   # Returns losses and detections
print(output)
``` 
the result is as follows:
```
({'boxes': tensor([[194., 214., 444., 293.]]), 'labels': tensor([1]), 'keypoints': tensor([[[234.,  53.,   1.],
         [209.,  46.,   1.],
         [ 38.,  40.,   1.]]]), 'image_id': tensor([252]), 'area': tensor([19750.]), 'iscrowd': tensor([0])}, {'boxes': tensor([[170., 264., 450., 352.]]), 'labels': tensor([1]), 'keypoints': tensor([[[267.,  61.,   1.],
         [231.,  48.,   1.],
         [ 35.,  40.,   1.]]]), 'image_id': tensor([202]), 'area': tensor([24640.]), 'iscrowd': tensor([0])})
{'loss_classifier': tensor(0.7168, grad_fn=<NllLossBackward>), 'loss_box_reg': tensor(0.0301, grad_fn=<DivBackward0>), 'loss_keypoint': tensor(-0., grad_fn=<MulBackward0>), 'loss_objectness': tensor(0.0251, grad_fn=<BinaryCrossEntropyWithLogitsBackward>), 'loss_rpn_box_reg': tensor(0.0051, grad_fn=<DivBackward0>)}
```
when I train the model,  output is like this:
```
Epoch: [0]  [  0/159]  eta: 0:03:31  lr: 0.000037  loss: 0.6780 (0.6780)  loss_classifier: 0.5869 (0.5869)  loss_box_reg: 0.0379 (0.0379)  loss_keypoint: -0.0000 (0.0000)  loss_objectness: 0.0429 (0.0429)  loss_rpn_box_reg: 0.0102 (0.0102)  time: 1.3282  data: 0.1701  max mem: 1835
Epoch: [0]  [ 10/159]  eta: 0:01:07  lr: 0.000353  loss: 0.2776 (0.3383)  loss_classifier: 0.0930 (0.2237)  loss_box_reg: 0.0906 (0.0936)  loss_keypoint: -0.0000 (0.0000)  loss_objectness: 0.0036 (0.0134)  loss_rpn_box_reg: 0.0078 (0.0076)  time: 0.4510  data: 0.0216  max mem: 2841
Epoch: [0]  [ 20/159]  eta: 0:00:55  lr: 0.000669  loss: 0.1753 (0.2336)  loss_classifier: 0.0870 (0.1453)  loss_box_reg: 0.0604 (0.0724)  loss_keypoint: -0.0000 (0.0000)  loss_objectness: 0.0009 (0.0074)  loss_rpn_box_reg: 0.0079 (0.0085)  time: 0.3553  data: 0.0068  max mem: 2841
Epoch: [0]  [ 30/159]  eta: 0:00:48  lr: 0.000985  loss: 0.1216 (0.7496)  loss_classifier: 0.0397 (0.1221)  loss_box_reg: 0.0330 (0.0586)  loss_keypoint: -0.0000 (0.5501)  loss_objectness: 0.0009 (0.0093)  loss_rpn_box_reg: 0.0086 (0.0095)  time: 0.3350  data: 0.0069  max mem: 2841
Epoch: [0]  [ 40/159]  eta: 0:00:42  lr: 0.001301  loss: 0.1281 (0.8739)  loss_classifier: 0.0497 (0.1094)  loss_box_reg: 0.0281 (0.0523)  loss_keypoint: -0.0000 (0.6869)  loss_objectness: 0.0062 (0.0153)  loss_rpn_box_reg: 0.0076 (0.0100)  time: 0.3170  data: 0.0069  max mem: 2841
Epoch: [0]  [ 50/159]  eta: 0:00:38  lr: 0.001617  loss: 0.1619 (0.7393)  loss_classifier: 0.0854 (0.1077)  loss_box_reg: 0.0302 (0.0530)  loss_keypoint: -0.0000 (0.5522)  loss_objectness: 0.0123 (0.0168)  loss_rpn_box_reg: 0.0076 (0.0097)  time: 0.3116  data: 0.0067  max mem: 2841
Epoch: [0]  [ 60/159]  eta: 0:00:33  lr: 0.001933  loss: 0.1954 (0.6523)  loss_classifier: 0.0981 (0.1058)  loss_box_reg: 0.0438 (0.0532)  loss_keypoint: -0.0000 (0.4617)  loss_objectness: 0.0204 (0.0205)  loss_rpn_box_reg: 0.0079 (0.0111)  time: 0.3083  data: 0.0063  max mem: 2841
Epoch: [0]  [ 70/159]  eta: 0:00:30  lr: 0.002250  loss: 0.1888 (0.6052)  loss_classifier: 0.1071 (0.1206)  loss_box_reg: 0.0388 (0.0568)  loss_keypoint: -0.0000 (0.3966)  loss_objectness: 0.0149 (0.0200)  loss_rpn_box_reg: 0.0109 (0.0112)  time: 0.3072  data: 0.0064  max mem: 2841
Epoch: [0]  [ 80/159]  eta: 0:00:26  lr: 0.002566  loss: 0.2676 (40.1674)  loss_classifier: 0.1194 (0.1981)  loss_box_reg: 0.0523 (0.0774)  loss_keypoint: -0.0000 (39.8500)  loss_objectness: 0.0131 (0.0275)  loss_rpn_box_reg: 0.0122 (0.0143)  time: 0.3032  data: 0.0068  max mem: 2841
Epoch: [0]  [ 90/159]  eta: 0:00:22  lr: 0.002882  loss: 1.5832 (286.2538)  loss_classifier: 0.4849 (4.9125)  loss_box_reg: 0.1978 (231.4146)  loss_keypoint: -0.0000 (35.4709)  loss_objectness: 0.2688 (9.5655)  loss_rpn_box_reg: 0.1115 (4.8903)  time: 0.2946  data: 0.0069  max mem: 2841
Epoch: [0]  [100/159]  eta: 0:00:19  lr: 0.003198  loss: 117.8058 (1639.4883)  loss_classifier: 2.3041 (364.9840)  loss_box_reg: 0.8040 (1207.7777)  loss_keypoint: 0.0000 (31.9589)  loss_objectness: 3.5009 (21.0504)  loss_rpn_box_reg: 2.0930 (13.7174)  time: 0.2902  data: 0.0071  max mem: 2841
Epoch: [0]  [110/159]  eta: 0:00:15  lr: 0.003514  loss: 806.4288 (9284.8999)  loss_classifier: 87.7771 (1006.4765)  loss_box_reg: 559.2792 (8211.0337)  loss_keypoint: 0.0000 (29.0797)  loss_objectness: 5.2868 (20.3284)  loss_rpn_box_reg: 7.0411 (17.9817)  time: 0.2757  data: 0.0067  max mem: 2841
Epoch: [0]  [120/159]  eta: 0:00:12  lr: 0.003830  loss: 3678.1040 (217528.5675)  loss_classifier: 0.0000 (1058.8272)  loss_box_reg: 3454.0095 (80028.7827)  loss_keypoint: 0.0000 (136296.2054)  loss_objectness: 24.5404 (87.6677)  loss_rpn_box_reg: 31.4709 (57.0789)  time: 0.2667  data: 0.0063  max mem: 2841
Epoch: [0]  [130/159]  eta: 0:00:09  lr: 0.004146  loss: 61700.0156 (213160.6617)  loss_classifier: 0.0000 (978.0007)  loss_box_reg: 61620.0312 (86112.3690)  loss_keypoint: 0.0000 (125891.9149)  loss_objectness: 93.1087 (101.8168)  loss_rpn_box_reg: 105.1428 (76.5552)  time: 0.2706  data: 0.0069  max mem: 2841
Epoch: [0]  [140/159]  eta: 0:00:05  lr: 0.004463  loss: 101101.1328 (250612.3410)  loss_classifier: 0.0000 (3227.6938)  loss_box_reg: 99932.5156 (129892.4495)  loss_keypoint: 0.0000 (116963.4103)  loss_objectness: 118.7737 (298.2073)  loss_rpn_box_reg: 218.8433 (230.5746)  time: 0.2667  data: 0.0069  max mem: 2841
Epoch: [0]  [150/159]  eta: 0:00:02  lr: 0.004779  loss: 262296.8438 (386757.2148)  loss_classifier: 0.0000 (3376.3291)  loss_box_reg: 255827.7344 (272176.4407)  loss_keypoint: 0.0000 (109217.4891)  loss_objectness: 313.9008 (910.0859)  loss_rpn_box_reg: 934.3903 (1076.8679)  time: 0.2633  data: 0.0067  max mem: 2841
Epoch: [0]  [158/159]  eta: 0:00:00  lr: 0.005000  loss: 358287.3125 (1380410.8749)  loss_classifier: 0.0000 (3206.4509)  loss_box_reg: 356983.3750 (1267741.9229)  loss_keypoint: 0.0000 (103722.2695)  loss_objectness: 1518.1926 (2782.3766)  loss_rpn_box_reg: 1316.3130 (2957.8822)  time: 0.2614  data: 0.0066  max mem: 2841
``` 
after one epoch, the loss of classifier is always 0:
```
Epoch: [2]  [  0/159]  eta: 0:01:08  lr: 0.005000  loss: 210822982795264.0000 (210822982795264.0000)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 210277790384128.0000 (210277790384128.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 70683779072.0000 (70683779072.0000)  loss_rpn_box_reg: 474517012480.0000 (474517012480.0000)  time: 0.4283  data: 0.1865  max mem: 2841
Epoch: [2]  [ 10/159]  eta: 0:00:41  lr: 0.005000  loss: 322171994112.0000 (129149086344576.0000)  loss_classifier: 0.0000 (9503493841.4598)  loss_box_reg: 206106984448.0000 (128987444614586.2344)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 172261040.0000 (18398419619.9545)  loss_rpn_box_reg: 989684160.0000 (133738707256.0000)  time: 0.2793  data: 0.0225  max mem: 2841
Epoch: [2]  [ 20/159]  eta: 0:00:37  lr: 0.005000  loss: 367935160320.0000 (196915458566686.4688)  loss_classifier: 0.0000 (4978020583.6286)  loss_box_reg: 365838336000.0000 (195495447054811.4375)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 1342751744.0000 (1077955654125.5000)  loss_rpn_box_reg: 3287106816.0000 (337079566583.8095)  time: 0.2628  data: 0.0065  max mem: 2841
Epoch: [2]  [ 30/159]  eta: 0:00:33  lr: 0.005000  loss: 11212734595072.0000 (592405466766472.2500)  loss_classifier: 0.0000 (3372207492.1355)  loss_box_reg: 11200620396544.0000 (591393494845101.3750)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 4489793024.0000 (738452164266.3347)  loss_rpn_box_reg: 25943838720.0000 (270164651896.3871)  time: 0.2542  data: 0.0064  max mem: 2841
Epoch: [2]  [ 40/159]  eta: 0:00:31  lr: 0.005000  loss: 29217811070976.0000 (25920732290451173376.0000)  loss_classifier: 0.0000 (2549717859.9074)  loss_box_reg: 28668816523264.0000 (493680713470745.0000)  loss_keypoint: 0.0000 (25920237669679185920.0000)  loss_objectness: 1187935488.0000 (592036092538.1555)  loss_rpn_box_reg: 12114538496.0000 (256035175177.8537)  time: 0.2538  data: 0.0058  max mem: 2841
Epoch: [2]  [ 50/159]  eta: 0:00:28  lr: 0.005000  loss: 8962653028352.0000 (167640256742106234880.0000)  loss_classifier: 0.0000 (2049773181.4941)  loss_box_reg: 8954078822400.0000 (413623601411970.5000)  loss_keypoint: -0.0000 (167639842833093689344.0000)  loss_objectness: 1744851712.0000 (478291576051.7720)  loss_rpn_box_reg: 45173231616.0000 (294271834978.2745)  time: 0.2551  data: 0.0056  max mem: 2841
Epoch: [2]  [ 60/159]  eta: 0:00:25  lr: 0.005000  loss: 15166388305920.0000 (140158266983234076672.0000)  loss_classifier: 0.0000 (1713744791.0853)  loss_box_reg: 15162222313472.0000 (365237905577543.3750)  loss_keypoint: -0.0000 (140157901385045540864.0000)  loss_objectness: 3792286464.0000 (407623597396.2029)  loss_rpn_box_reg: 34348660736.0000 (359950220193.9016)  time: 0.2589  data: 0.0059  max mem: 2841
Epoch: [2]  [ 70/159]  eta: 0:00:23  lr: 0.005000  loss: 47605258125312.0000 (120421281683919765504.0000)  loss_classifier: 0.0000 (1472372285.2986)  loss_box_reg: 47363905290240.0000 (3922998060708298.0000)  loss_keypoint: -0.0000 (120417351894194044928.0000)  loss_objectness: 3792286464.0000 (1831438876405.3574)  loss_rpn_box_reg: 100847026176.0000 (5310219475512.1689)  time: 0.2652  data: 0.0064  max mem: 2841
Epoch: [2]  [ 80/159]  eta: 0:00:20  lr: 0.005000  loss: 98666547249152.0000 (105778840047690924032.0000)  loss_classifier: 0.0000 (1290597929.0889)  loss_box_reg: 95200550584320.0000 (3590908700510028.0000)  loss_keypoint: -0.0000 (105775243107687448576.0000)  loss_objectness: 6724773376.0000 (1619820040814.7700)  loss_rpn_box_reg: 300744245248.0000 (4728634353811.2100)  time: 0.2577  data: 0.0066  max mem: 2841
Epoch: [2]  [ 90/159]  eta: 0:00:17  lr: 0.005000  loss: 98666547249152.0000 (94154818875704295424.0000)  loss_classifier: 0.0000 (1148773980.8374)  loss_box_reg: 95200550584320.0000 (3223440291824018.5000)  loss_keypoint: -0.0000 (94151590018930589696.0000)  loss_objectness: 6724773376.0000 (1466143453208.0481)  loss_rpn_box_reg: 97111777280.0000 (4232637918621.7583)  time: 0.2517  data: 0.0061  max mem: 2841
Epoch: [2]  [100/159]  eta: 0:00:15  lr: 0.005000  loss: 28301326286848.0000 (84832566141457022976.0000)  loss_classifier: 0.0000 (1035033982.7350)  loss_box_reg: 28299304632320.0000 (2910844112899607.0000)  loss_keypoint: 0.0000 (84829650413095878656.0000)  loss_objectness: 911813952.0000 (1321187310969.7068)  loss_rpn_box_reg: 13324769280.0000 (3817410141258.9702)  time: 0.2584  data: 0.0061  max mem: 2841
Epoch: [2]  [110/159]  eta: 0:00:12  lr: 0.005000  loss: 15452155674624.0000 (77190006224720429056.0000)  loss_classifier: 0.0000 (941787677.9843)  loss_box_reg: 15438115241984.0000 (2662211871198270.5000)  loss_keypoint: 0.0000 (77187339565069221888.0000)  loss_objectness: 339361728.0000 (1202497745570.9944)  loss_rpn_box_reg: 6879087616.0000 (3476717937819.8560)  time: 0.2708  data: 0.0061  max mem: 2841
Epoch: [2]  [120/159]  eta: 0:00:10  lr: 0.005000  loss: 15452155674624.0000 (70810683756778856448.0000)  loss_classifier: 0.0000 (863953985.5893)  loss_box_reg: 15438115241984.0000 (2459062950763966.5000)  loss_keypoint: 0.0000 (70808220592749453312.0000)  loss_objectness: 859669504.0000 (1109014496474.8130)  loss_rpn_box_reg: 7434253312.0000 (3204373769526.1157)  time: 0.2751  data: 0.0066  max mem: 2841
Epoch: [2]  [130/159]  eta: 0:00:07  lr: 0.005000  loss: 5228529713152.0000 (65405351195009007616.0000)  loss_classifier: 0.0000 (798003299.6668)  loss_box_reg: 5224993390592.0000 (2334473663616412.5000)  loss_keypoint: 0.0000 (65403012913913618432.0000)  loss_objectness: 859669504.0000 (1030950916137.2090)  loss_rpn_box_reg: 2676395008.0000 (2972585382236.4888)  time: 0.2663  data: 0.0070  max mem: 2841
Epoch: [2]  [140/159]  eta: 0:00:04  lr: 0.005000  loss: 10647136894976.0000 (60766686378433191936.0000)  loss_classifier: 0.0000 (741407320.9670)  loss_box_reg: 10640270819328.0000 (2181467213309732.2500)  loss_keypoint: 0.0000 (60764501359735349248.0000)  loss_objectness: 11241362432.0000 (959576818407.6481)  loss_rpn_box_reg: 43245166592.0000 (2774104026177.9575)  time: 0.2541  data: 0.0068  max mem: 2841
Epoch: [2]  [150/159]  eta: 0:00:02  lr: 0.005000  loss: 172357373132800.0000 (72428027274601349120.0000)  loss_classifier: 0.0000 (3494113203.7115)  loss_box_reg: 136885808660480.0000 (3297685031823996.0000)  loss_keypoint: 0.0000 (72424725506553561088.0000)  loss_objectness: 13322499072.0000 (899359951511.1946)  loss_rpn_box_reg: 99951337472.0000 (2610520990204.3179)  time: 0.2637  data: 0.0069  max mem: 2841
Epoch: [2]  [158/159]  eta: 0:00:00  lr: 0.005000  loss: 665828043259904.0000 (68797526438743728128.0000)  loss_classifier: 0.0000 (3318308765.7893)  loss_box_reg: 495787268112384.0000 (16777376688624388.0000)  loss_keypoint: 0.0000 (68780714160311877632.0000)  loss_objectness: 47107575808.0000 (8415982953399.8398)  loss_rpn_box_reg: 228761092096.0000 (25941666767845.9609)  time: 0.2711  data: 0.0064  max mem: 2841
```
what is more, in evluation, the issue https://github.com/pytorch/vision/issues/1568#issuecomment-606989009  occurred, and was solved by the mentioned code.

What is wrong for training a model with 3 keypoints with my dataset?

**Comment by vincentqb** [comment]
This issue is better discussed on the [forum](https://discuss.pytorch.org). I will close this issue, but please feel free to re-open.
